callbacks/callmethod.go

```diff
@@ -17,6 +17,9 @@ func callMethod(db *gorm.DB, fc func(value interface{}, tx *gorm.DB) bool) {
 				db.Statement.CurDestIndex++
 			}
 		case reflect.Struct:
+			if !db.Statement.ReflectValue.CanAddr() {
+				db.Statement.ReflectValue = reflect.New(db.Statement.ReflectValue.Type()).Elem()
```

Comment:
I think maybe better to raise an error?

Reply:
Do you mean that param of `Model()` method must be ptr?